flink-libraries/flink-table/pom.xml

```diff
@@ -213,36 +209,49 @@ under the License.
 								<filter>
 									<artifact>*:*</artifact>
 									<excludes>
+										<!-- excluded all these files for a clean flink-table jar -->
 										<exclude>org-apache-calcite-jdbc.properties</exclude>
+										<exclude>common.proto</exclude>
+										<exclude>requests.proto</exclude>
+										<exclude>responses.proto</exclude>
 										<exclude>mozilla/**</exclude>
 										<exclude>codegen/**</exclude>
 										<exclude>google/**</exclude>
 										<exclude>META-INF/*.SF</exclude>
 										<exclude>META-INF/*.DSA</exclude>
 										<exclude>META-INF/*.RSA</exclude>
+										<exclude>META-INF/services/**</exclude>
+										<exclude>properties.dtd</exclude>
+										<exclude>PropertyList-1.0.dtd</exclude>
+										<exclude>digesterRules.xml</exclude>
+										<!-- not excluded for now -->
+										<!-- <exclude>org.codehaus.commons.compiler.properties</exclude> -->
 									</excludes>
 								</filter>
 							</filters>
 							<artifactSet>
 								<includes combine.children="append">
+									<!-- Calcite and its dependencies -->
 									<include>org.apache.calcite:*</include>
 									<include>org.apache.calcite.avatica:*</include>
+									<include>com.google.guava:guava</include>
+									<include>org.eigenbase:*</include>
 									<include>net.hydromatic:*</include>
-									<include>org.reflections:*</include>
+
+									<!-- flink-table dependencies -->
+									<include>commons-configuration:*</include>
+									<include>commons-lang:*</include>
+									<include>commons-codec:*</include>
 									<include>org.codehaus.janino:*</include>
-									<include>com.google.guava:guava</include>
-                                </includes>
+									<include>org.reflections:*</include>
+									<include>joda-time:*</include>
+								</includes>
 							</artifactSet>
 							<relocations>
-								<!-- We currently don't relocate slf4j as we have "logger not found" 
-									warnings otherwise during runtime -->
-								<!--<relocation>
-									<pattern>org.slf4j</pattern>
-									<shadedPattern>org.apache.flink.shaded.calcite.org.slf4j</shadedPattern>
-								</relocation>-->
+								<!-- Calcite and its dependencies -->
 								<relocation>
-									<pattern>com.fasterxml.jackson</pattern>
-									<shadedPattern>org.apache.flink.shaded.calcite.com.fasterxml.jackson</shadedPattern>
+									<pattern>org.apache.calcite</pattern>
+									<shadedPattern>org.apache.flink.shaded.calcite.org.apache.calcite</shadedPattern>
```

Comment:
For calcite dependencies, please change the shading pattern to org.apache.flink.calcite.shaded to a) make them more unique and b) distinguish them from flink-shaded dependencies.